### PR TITLE
Matthew week 2 HW attempt

### PR DESCRIPTION
**Pull request by mtbdata711**

### CODE REVIEW

File: week-2-practise/level-1/5-indexOf.js [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 var itemToFind = 7;
 var arr = [3, 5, 61, 7, 123];
-var index; // ONLY EDIT THIS LINE
+var index = arr.indexOf(itemToFind); // ONLY EDIT THIS LINE
```
Comment (clems27) [inline week-2-practise/level-1/5-indexOf.js]: Nice expression, well done

File: week-2-practise/level-3/6-map.js [hunk 2]
```diff
@@ -19,14 +19,11 @@ function capitalise(str) {
...
 var mentors = ["/Daniel ", "irina ", " Gordon", "ashleigh "];
 
-var mentorsMapped = mentors.map(tidyUpString);
-
-var mentorsTidyAndCapitalised = [];
-
-mentorsMapped.forEach(function(element) {
-  var change = capitalise(element);
-  mentorsTidyAndCapitalised.push(change);
+var mentorsTidyAndCapitalised = mentors.map(function(str) {
```
Comment (Atayo) [inline week-2-practise/level-3/6-map.js]: Great piece of work Matthew, well done.

File: week-2-practise/level-3/5-indexOf.js [hunk 3]
```diff
@@ -4,9 +4,22 @@
...
 // - return a new array with the replacement value inserted
+// Matthew comment: bit hacky. forEach and forLoop are combined so that any looping through the array
+// ends at the first index.
 
 function replace(arr, valueToReplace, newValue) {
-  return; // complete this statement
+  var newArray = [];
+  arr.forEach(function(item) {
+    newArray.push(item);
+  });
```
Comment (theseafaringturtle) [inline week-2-practise/level-3/5-indexOf.js]: same tip, newArray = arr.slice(0) is faster to write

File: week-2-practise/level-3/5-indexOf.js [hunk 4]
```diff
@@ -4,9 +4,22 @@
...
+  var getIndex = newArray.indexOf(valueToReplace);
+  for (var i = 0; i < newArray.length; i++) {
+    if (i === getIndex) {
+      newArray[i] = newValue;
+    }
+  }
```
Comment (theseafaringturtle) [inline week-2-practise/level-3/5-indexOf.js]: `newArray[getIndex] = newValue;` would do the same thing as the for loop

**Pull request closed without merging**